### ISSUE TITLE
Prallel Execution of Nodes

### ISSUE THREAD

**Issue by ChrisDelClea** [VinciGit00/Scrapegraph-ai#199]
Hey Folks, 

this looks like a great project. I am not sure if its a bug or a feature. 

**Describe the bug**
>  Executes the node's logic to instantiate and run multiple graph instances in parallel.
Stated in [here](https://github.com/VinciGit00/Scrapegraph-ai/blob/main/scrapegraphai/nodes/graph_iterator_node.py#L11)

**To Reproduce**
Can someone explain to me, where the parallelism kicks in? As far as i can tell, it runs in a for loop: 
```python
        # run the graph for each url and use tqdm for progress bar
        graphs_answers = []
        for graph in tqdm(graphs_instances, desc="Processing Graph Instances", disable=not self.verbose):
            result = graph.run()
            graphs_answers.append(result)
```

**Expected behavior**
I would expect that the nodes are executed in parallel and joined back together for speedup .

**Comment by PeriniM** [comment]
hey @ChrisDelClea Chris, yes it is a feature to implement! Right now it is iterating over the SmartScraper instances so it is sequential but we could implement an `async`/`await` mechanism on the run method of the graphs

**Comment by DiTo97** [comment]
@PeriniM I might take care of it as soon as I'm done running tests for #147, should be easy to make it asynchronous.

I would [use a semaphore](https://github.com/langchain-ai/langchain/blob/master/libs/community/langchain_community/document_loaders/async_html.py#L173) with a param specifying the number of concurrent graphs, not to exhaust too many resources.

**Comment by PeriniM** [comment]
@DiTo97 That would be sweet!

**Comment by DiTo97** [comment]
@PeriniM opened a PR; code should be working, but better if you test it first hand before merging

**Comment by PeriniM** [comment]
Closed with #213
